src/App.tsx

```diff
@@ -15,35 +18,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  default = '',
+  length = 'length',
+  alphabetically = 'alphabetically',
+  reverse = 'reverse',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabetically:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverseField ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const toReverse = () => {
+    if (!isReversed) {
+      return setIsReversed(true);
+    } else {
+      return setIsReversed(false);
+    }
+  };
```

Comment:
```suggestion
  const toReverse = () => setIsReversed(prevIsReversed => !prevIsReversed);
```

it can be shorter and better to change prev isRevered state in this case.